### DISCUSSION TITLE
Instcalc Lost My Previous Calculations

### DISCUSSION THREAD

**Discussion by dbareis** [kazad/instacalc#77] [Ideas]
I had previously logged on and saved some sheets.  On A different computer I used it without logging on.  I now logon and my sheets are not there and the updates I made to the sheet while not logged on is!

**Comment by kazad** [comment]
Hi @dbareis, sorry to hear about that! InstaCalc tries to reconcile the changes, I'd like to make this more clear when the latest version will override the previous. I have backups for the calcs, if you give me the name of the calc and the date you want (e.g., Mar 19, 2024), I can send you the calc data as of that date.

Thanks for using the site!

**Comment by dbareis** [comment]
Hi @kazad , I really like your tool as it is very similar to one of the few Windows programs I ever bought (OpalCalc) but I don't want clarification, I want it to never occur ever.  But, if I can't rely on it keeping past calculations then I won't use it.  It needs to merge changes (or keep existing).

I have shared one of the calc sheets and now it goes nowhere...

As for backup, why don't I have that access?  I just want it to be the way it was before it clobbered things yesterday, my email address is same as my id AT gmail & com.

**Comment by kazad** [comment]
Glad you like it! OpalCalc and other text-based scratchpads are great. Unfortunately, the sync/backup behavior is built into the service provider (Firebase). The backups are done on a daily basis, but there isn't a user-facing way to restore or choose the sync overwrite behavior, generally it's last-write wins. I'd like to see if I can improve this UX client-side.

Can you share the link to the shared calc in question? I see https://instacalc.com/57282, which looks to be the same as it was a few days ago in the backup. If that's the calc, I can get you the backup as of any date.

**Reply by dbareis** [reply]
I can see you restored it thanks :-)

I don't know about firebase but if it caused that issue I'd be changing to something else.  At the minimum we need a export and import option, I can export of any major changes...

**Reply by kazad** [reply]
Glad everything's as you need it! Yes, definitely thinking about an import/export option, it's on my list :).
